File: Api/DataAccess/Entities/SubscriptionPlanEntity.cs
namespace ReportChecker.DataAccess.Entities;

public class SubscriptionPlanEntity
{
    public required Guid Id { get; init; }
    public required string Name { get; init; }
    public string? Description { get; init; }
    public int TokensLimit { get; init; }
    public int ReportsLimit { get; init; }
    public bool IsHidden { get; init; }

    public required DateTime CreatedAt { get; init; }
    public DateTime? DeletedAt { get; init; }

    public virtual ICollection<SubscriptionOfferEntity> Offers { get; init; } = [];
    public virtual ICollection<UserSubscriptionEntity> UserSubscriptions { get; init; } = [];
}

File: ReportChecker/src/app/entities/subscription-plan-entity.ts
import {Moment} from 'moment';

export interface SubscriptionPlanEntity {
  id: string;
  name?: string;
  description?: string;
  tokensLimit: number;
  reportsLimit: number;
  createdAt?: Moment;
  deletedAt?: Moment;
  offers: SubscriptionOfferEntity[];
}

export interface SubscriptionOfferEntity {
  id: string;
  planId: string;
  price: number;
  months: number;
  createdAt?: Moment;
  deletedAt?: Moment;
}
